casper/src/test/scala/coop/rchain/casper/addblock/MultiParentCasperAddBlockSpec.scala

```diff
@@ -498,21 +498,22 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
   it should "succeed at slashing" in effectTest {
     TestNode.networkEff(genesis, networkSize = 3).use { nodes =>
       for {
-        deployData   <- ConstructDeploy.basicDeployData[Effect](0)
-        signedBlock  <- nodes(0).casperEff.deploy(deployData) >> nodes(0).createBlockUnsafe()
-        invalidBlock = signedBlock.copy(seqNum = 47)
-        status1      <- nodes(1).processBlock(invalidBlock)
-        status2      <- nodes(2).processBlock(invalidBlock)
-        signedBlock2 <- nodes(1).createBlockUnsafe()
-        status3      <- nodes(1).processBlock(signedBlock2)
-        bonds        <- nodes(1).runtimeManager.computeBonds(ProtoUtil.postStateHash(signedBlock2))
-        _            = bonds.map(_.stake).min should be(0) // Slashed validator has 0 stake
-        _            <- nodes(2).handleReceive()
-        signedBlock3 <- nodes(2).createBlockUnsafe()
-        status4      <- nodes(2).processBlock(signedBlock3)
+        deployData    <- ConstructDeploy.basicDeployData[Effect](0, shardId = SHARD_ID)
```

Comment:
It seems that shard was left out here so your fix in added commit looks legitimate.

Reply:
I meant that there are some places in this file where `basicDeployData` is called, but `SHARD_ID` is not filled. To avoid possible problems, maybe it's worth adding the `SHARD_ID` filling there and doing it also for the `v0.12` branch?